tests/test_speckles.py

```diff
@@ -0,0 +1,247 @@
+import numpy as np
+
+from lasy.laser import Laser
+from lasy.profiles.speckle_profile import SpeckleProfile
+import pytest
+from scipy.constants import c
+
```

Comment:
For each of the `test_...` function, could you add one or two sentence to explain what this test checks? (potentially with a reference to the corresponding section of P. Michel's book?

Reply:
Yes, but I can't add the section references until I am back in town

Reply:
Done